react/week2/to_do_list/src/index.js

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const todos = [
+  {
+    id: 0,
+    description: "Get out of bed",
+    color: "brown",
+    deadline: "2020-06-08",
+    done: false,
+    Visibility: true
+  },
+  {
+    id: 1,
+    description: "Brush teeth",
+    color: "red",
+    deadline: "2020-06-08",
+    done: false,
+    Visibility: true
+  },
+  {
+    id: 2,
+    description: "Eat breakfast",
+    color: "orange",
+    deadline: "2020-06-08",
+    done: true,
+    Visibility: true
+  }
+];
+
+function getRandomColor() {
+  var letters = "0123456789ABCDEF";
+  var color = "#";
+  for (var i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+}
+
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { renderTrigger: true };
+  }
+
+  deleteTask(index) {
+    this.props.taskData[index].Visibility = false;
+    this.setState({ renderTrigger: !this.state.renderTrigger });
+  }
+
+  taggleDoneState(index) {
+    this.props.taskData[index].done = !this.props.taskData[index].done;
```

Comment:
props are read only, you can't mutate them: https://reactjs.org/docs/components-and-props.html#props-are-read-only